Apsitvarkom.Api/Controllers/PollutedLocationController.cs

```diff
@@ -71,7 +84,41 @@ public async Task<ActionResult<PollutedLocationGetResponse>> GetById(string id)
         try
         {
             var location = await _repository.GetByPropertyAsync(x => x.Id.ToString() == id);
-            return location is not null ? Ok(location) : NotFound($"Polluted location with the specified id '{id}' was not found.");
+
+            if (location is null) return NotFound($"Polluted location with the specified id '{id}' was not found.");
+
+            var mappedLocation = _mapper.Map<PollutedLocationGetResponse>(location);
+            if (mappedLocation is null) return StatusCode(StatusCodes.Status500InternalServerError);
+
+            return Ok(mappedLocation);
+        }
+        catch (Exception)
+        {
+            return StatusCode(StatusCodes.Status500InternalServerError);
+        }
+    }
+
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(List<string>))]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    [HttpPost("Create")]
+    public async Task<ActionResult<PollutedLocationGetResponse>> Create(PollutedLocationCreateRequest pollutedLocationCreateRequest)
+    {
+        var validationResult = await _pollutedLocationValidator.ValidateAsync(pollutedLocationCreateRequest);
+        if (!validationResult.IsValid) return BadRequest(validationResult.Errors.Select(e => e.ErrorMessage).ToList());
+
+        var mappedPollutedLocation = _mapper.Map<PollutedLocation>(pollutedLocationCreateRequest);
+        if (mappedPollutedLocation is null) return StatusCode(StatusCodes.Status500InternalServerError);
+
+        try
+        {
+            var result = await _repository.InsertAsync(mappedPollutedLocation);
+
+            var response = _mapper.Map<PollutedLocationGetResponse>(result);
+            // If failed to map the response, still return 201 because the location was inserted into the database
```

Comment:
If we can't map it to response, there must have been something wrong with the data I believe. Shouldn't we delete the inserted location in this case (would require #90) instead of potentially having it junked with wrong data?

Reply:
this would also make it so that the same status code can return two completely different responses content wise.

Reply:
See this convo: https://github.com/vu-vibedosa/apsitvarkom-be/pull/101/#discussion_r1027133864